src/plugins/dashboard/public/dashboard_api/get_serialized_state.test.ts

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import type { DashboardPanelState } from '../../common';
+
+import {
+  dataService,
+  embeddableService,
+  savedObjectsTaggingService,
+} from '../services/kibana_services';
+import { getSampleDashboardState } from '../mocks';
+import { getSerializedState } from './get_serialized_state';
+
+dataService.search.searchSource.create = jest.fn().mockResolvedValue({
+  setField: jest.fn(),
+  getSerializedFields: jest.fn().mockReturnValue({}),
+});
+
+dataService.query.timefilter.timefilter.getTime = jest
+  .fn()
+  .mockReturnValue({ from: 'now-15m', to: 'now' });
+
+dataService.query.timefilter.timefilter.getRefreshInterval = jest
+  .fn()
+  .mockReturnValue({ pause: true, value: 0 });
+
+embeddableService.extract = jest
+  .fn()
+  .mockImplementation((attributes) => ({ state: attributes, references: [] }));
+
+if (savedObjectsTaggingService) {
+  savedObjectsTaggingService.getTaggingApi = jest.fn().mockReturnValue({
+    ui: {
+      updateTagsReferences: jest.fn((references, tags) => references),
+    },
+  });
+}
+
+jest.mock('uuid', () => ({
+  v4: jest.fn().mockReturnValue('54321'),
+}));
+
+describe('getSerializedState', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should return the current state attributes and references', async () => {
+    const dashboardState = getSampleDashboardState();
+    const result = await getSerializedState({
```

Comment:
Calls to `getSerializedState` should not be async. Also, all test cases do not need to be async